lib/php/rendition-factory/src/Transformer/Video/SummaryTransformerModule.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+namespace Alchemy\RenditionFactory\Transformer\Video;
+
+use Alchemy\RenditionFactory\DTO\FamilyEnum;
+use Alchemy\RenditionFactory\DTO\InputFileInterface;
+use Alchemy\RenditionFactory\DTO\OutputFile;
+use Alchemy\RenditionFactory\DTO\OutputFileInterface;
+use Alchemy\RenditionFactory\Transformer\TransformationContext;
+use Alchemy\RenditionFactory\Transformer\TransformerModuleInterface;
+use FFMpeg;
+use FFMpeg\Coordinate\TimeCode;
+use FFMpeg\Format\FormatInterface;
+
+final readonly class SummaryTransformerModule implements TransformerModuleInterface
+{
+    public function __construct()
+    {
+    }
+
+    public static function getName(): string
+    {
+        return 'summary';
+    }
+
+    public function transform(InputFileInterface $inputFile, array $options, TransformationContext $context): OutputFileInterface
+    {
+        if (!($format = $options['format'])) {
+            throw new \InvalidArgumentException('Missing format');
+        }
+        if (!($extension = $options['extension'])) {
+            throw new \InvalidArgumentException('Missing extension');
+        }
+
+        $fqcnFormat = 'FFMpeg\\Format\\Video\\'.$format;
+        if (class_exists($fqcnFormat)) {
+            return $this->doVideo($format, $extension, $inputFile, $options, $context);
+        }
+
+        throw new \InvalidArgumentException(sprintf('Invalid format %s', $format));
+    }
+
+    private function doVideo(string $format, string $extension, InputFileInterface $inputFile, array $options, TransformationContext $context): OutputFileInterface
+    {
+        $period = $options['period'] ?? 0;
+        if ($period <= 0) {
+            throw new \InvalidArgumentException('Invalid period for module "summary"');
+        }
+        $clipDuration = $options['duration'] ?? 0;
+        if ($clipDuration <= 0 || $clipDuration >= $period) {
+            throw new \InvalidArgumentException('Invalid duration for module "summary"');
+        }
+
+        $fqcnFormat = 'FFMpeg\\Format\\Video\\'.$format;
+        $outpuFormat = new $fqcnFormat();
+        if ($videoCodec = $options['video_codec'] ?? null) {
+            if (!in_array($videoCodec, $outpuFormat->getAvailableVideoCodecs())) {
+                throw new \InvalidArgumentException(sprintf('Invalid video codec %s for format %s', $videoCodec, $format));
+            }
+            $outpuFormat->setVideoCodec($videoCodec);
+        }
+        if ($audioCodec = $options['audio_codec'] ?? null) {
+            if (!in_array($audioCodec, $outpuFormat->getAvailableAudioCodecs())) {
+                throw new \InvalidArgumentException(sprintf('Invalid audio codec %s for format %s', $audioCodec, $format));
+            }
+            $outpuFormat->setAudioCodec($audioCodec);
+        }
+
+        // try to find an "enhanced by alchemy" format (supports "copy" video codec) for clips
```

Comment:
"copy" video codec is strange. Why not using "same" or "keep_same"?

Reply:
`copy` is a real codec in ffmpeg

Reply:
:open_mouth: